src/evaluate.py

```diff
@@ -98,13 +98,35 @@ def evaluation(org_img_path, pred_img_path, mode, write_to_file):
             metric_dict[metric] = {'ISSM': issm_value}
             write_final_dict(metric, metric_dict)
 
+    if metric == 'uiq':
```

Comment:
This whole block can be condensed a lot by using the "exec" construct. Did not test it, but should look something like this (using f-strings btw):

out_value = exec(f"{metric}(org_img, pred_img)")
logger.info(f"{metric.upper()} value is: {out_value}")
if write_to_file:
    metric_dict[metric] = {'metric.upper()': out_value}
    write_final_dict(metric, metric_dict)

Reply:
Oh, I never used this, I will try it now, Thank you.